src/Middleware/WebSockets/src/HandshakeHelpers.cs

```diff
@@ -110,13 +123,25 @@ public static string CreateResponseKey(string requestKey)
                 throw new ArgumentNullException(nameof(requestKey));
             }
 
-            using (var algorithm = SHA1.Create())
+            if (_algorithm == null)
             {
-                string merged = requestKey + "258EAFA5-E914-47DA-95CA-C5AB0DC85B11";
-                byte[] mergedBytes = Encoding.UTF8.GetBytes(merged);
-                byte[] hashedBytes = algorithm.ComputeHash(mergedBytes);
-                return Convert.ToBase64String(hashedBytes);
+                _algorithm = SHA1.Create();
             }
+
+            // requestKey is already verified to be small (24 bytes) by 'IsRequestKeyValid()' and everything is 1:1 mapping to UTF8 bytes
+            // so this can be hardcoded to 60 bytes for the requestKey + static websocket string
+            Span<byte> mergedBytes = stackalloc byte[60];
+            Encoding.UTF8.GetBytes(requestKey, mergedBytes);
+            _encodedWebSocketKey.CopyTo(mergedBytes.Slice(24));
```

Comment:
At 24 + 36 bytes I don't know if it would be faster to use IncrementalHash with two calls to AppendData or to do the copy (probably doing this copy).

You might see perf improvement from using IncrementalHash anyways, if you happen to be at a level where you can notice a 3us difference.

Reply:
I'll check this out

Reply:
Tried it out and it looked slower